source/Calamari/Integration/FileSystem/CalamariPhysicalFileSystem.cs

```diff
@@ -335,41 +362,46 @@ public void EnsureDirectoryExists(string directoryPath)
 
         // ReSharper disable AssignNullToNotNullAttribute
 
-        public void CopyDirectory(string sourceDirectory, string targetDirectory, int overwriteFileRetryAttempts = 6)
+        public int CopyDirectory(string sourceDirectory, string targetDirectory)
         {
-            CopyDirectory(sourceDirectory, targetDirectory, CancellationToken.None, overwriteFileRetryAttempts);
+            return CopyDirectory(sourceDirectory, targetDirectory, CancellationToken.None);
         }
 
-        public void CopyDirectory(string sourceDirectory, string targetDirectory, CancellationToken cancel, int overwriteFileRetryAttempts = 6)
+        public int CopyDirectory(string sourceDirectory, string targetDirectory, CancellationToken cancel)
         {
             if (!DirectoryExists(sourceDirectory))
-                return;
+                return 0;
 
             if (!DirectoryExists(targetDirectory))
             {
                 Directory.CreateDirectory(targetDirectory);
             }
 
+            int count = 0;
             var files = Directory.GetFiles(sourceDirectory, "*");
             foreach (var sourceFile in files)
             {
                 cancel.ThrowIfCancellationRequested();
 
                 var targetFile = Path.Combine(targetDirectory, Path.GetFileName(sourceFile));
-                CopyFile(sourceFile, targetFile, overwriteFileRetryAttempts);
+                CopyFile(sourceFile, targetFile);
+                count++;
             }
 
             foreach (var childSourceDirectory in Directory.GetDirectories(sourceDirectory))
             {
                 var name = Path.GetFileName(childSourceDirectory);
                 var childTargetDirectory = Path.Combine(targetDirectory, name);
-                CopyDirectory(childSourceDirectory, childTargetDirectory, cancel, overwriteFileRetryAttempts);
+                count += CopyDirectory(childSourceDirectory, childTargetDirectory, cancel);
             }
```

Comment:
It could be, but a better place for an additional check might be above `Directory.CreateDirectory`. This loop does no  time consuming work of its own, it will quickly recurse back into the same method and as soon as it hits a file it will stop.